backend/amd64/regalloc_stack_operands.ml

```diff
@@ -181,8 +181,13 @@ let basic (map : spilled_map) (instr : Cfg.basic Cfg.instruction) =
     may_use_stack_operand_for_second_argument map instr
   | Op (Specific (Isimd op)) ->
     (match Simd_selection.register_behavior op with
-    | R_to_R | R_R_to_fst -> May_still_have_spilled_registers
-    | R_RM_to_fst -> may_use_stack_operand_for_second_argument map instr
+    | R_to_fst | R_to_R | R_R_to_fst -> May_still_have_spilled_registers
+    | R_RM_to_fst | String_no_length | String_no_length_mask ->
+      may_use_stack_operand_for_second_argument map instr
+    | String_length | String_length_mask ->
+      may_use_stack_operand_for_second_argument ~num_args:4 map instr
```

Comment:
Hmm, good catch. This is only used here + `Ifloat_min/max` (which are getting moved into the SIMD path), so I think it's okay to just remove that check.

Reply:
After discussing offline with @xclerc: `may_use_stack_operand_for_second_argument` is sometimes used for instructions that require `res.(0)` and `arg.(0)` to be the same, so it would be better to keep this check in some way (either another another argument, something like `~to_fst` or just separately checking this condition where needed). The code in this area is quite subtle; these invariants are possibly saving us debugging time down the road, esp as we experiement with other register allocators. This module is probably going to be refactored soon, to make this simpler to understand and more precise, and at that point we can reconsider the amount of extra checking we need here.

Reply:
Ok, restored the check with an extra parameter